store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/util/ConcatenateMergeFunctionTest.java

```diff
@@ -0,0 +1,113 @@
+ */
+
+package uk.gov.gchq.gaffer.federatedstore.util;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.STRING;
+
+class ConcatenateMergeFunctionTest {
+
+    @Test
+    public void shouldReturnEmptyListWhenUpdateAndStateNull() {
+        // Given
+        final ConcatenateMergeFunction mergeFunction = new ConcatenateMergeFunction();
+
+        // When
+        Iterable<Object> results = mergeFunction.apply(null, null);
+
+        // Then
+        assertThat(results)
+                .isNotNull()
+                .isEmpty();
+    }
+
+    @Test
+    public void shouldReturnStateWhenUpdateNull() {
+        // Given
+        final ConcatenateMergeFunction mergeFunction = new ConcatenateMergeFunction();
+
+        // When
+        Iterable<Object> state = Collections.singletonList(STRING);
+        Iterable<Object> results = mergeFunction.apply(null, state);
+
+        // Then
+        assertThat(results)
+                .containsExactlyElementsOf(state);
+    }
+
+    @Test
+    public void shouldReturnUpdateWhenStateNull() {
+        // Given
+        final ConcatenateMergeFunction mergeFunction = new ConcatenateMergeFunction();
+
+        // When
+        Iterable<Object> update = Collections.singletonList(STRING);
+        Iterable<Object> results = mergeFunction.apply(update, null);
+
+        // Then
+        assertThat(results)
+                .containsExactlyElementsOf(update);
+    }
+
+    @Test
+    public void shouldReturnUpdateWrappedInIterableWhenStateNull() {
+        // Given
+        final ConcatenateMergeFunction mergeFunction = new ConcatenateMergeFunction();
+
+        // When
+        Object update = STRING;
+        Iterable<Object> results = mergeFunction.apply(update, null);
+
+        // Then
+        assertThat(results)
+                .containsExactly(update);
+    }
+
+    @Test
+    public void shouldConcatenateTwoIterablesIntoOne() {
+        // Given
+        final ConcatenateMergeFunction mergeFunction = new ConcatenateMergeFunction();
+
+        // When
+        Iterable<Object> update = Collections.singletonList(STRING);
+        Iterable<Object> state = Collections.singletonList(STRING);
+        Iterable<Object> results = mergeFunction.apply(update, state);
+
+        // Then
+        assertThat(results)
+                .containsExactly(STRING, STRING);
+    }
+
+    @Test
+    public void shouldWrapUpdateInIterableAndConcatenate() {
+        // Given
+        final ConcatenateMergeFunction mergeFunction = new ConcatenateMergeFunction();
+
+        // When
+        Object update = STRING;
+        Iterable<Object> state = Collections.singletonList(STRING);
+        Iterable<Object> results = mergeFunction.apply(update, state);
+
+        // Then
+        assertThat(results)
+                .containsExactly(STRING, STRING);
+    }
+}
```

Comment:
Can you add a test for
update = iterable[null]
state = iterable[null]

assertThat(output).containsExactly(null, null)